### PR TITLE
`gluonts.itertools`: Add `Fuse`.

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/itertools.py [hunk 1]
```diff
@@ -140,12 +145,150 @@ class Chain:
...
+            assert step is None or step == 1
+
+            start_index = self._location_for(start)
+            stop_index = self._location_for(stop)
+            return self._get_range(start_index, stop_index)
+
+        subindex = self._location_for(idx)
+        return self.collections[subindex.item][subindex.local]
+
+    def split(self, indices: List[int]) -> List[Fuse]:
```
Comment (lostella) [inline src/gluonts/itertools.py]: This could be defined for generic collections maybe?

File: src/gluonts/itertools.py [hunk 2]
```diff
@@ -140,12 +145,150 @@ class Chain:
...
+        for stop in indices:
+            results.append(self[start:stop])
+
+            start = stop
+
+        results.append(self[start:])
+
+        return results
+
+    def split_into(self, n):
```
Comment (lostella) [inline src/gluonts/itertools.py]: Same, maybe

File: src/gluonts/itertools.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/itertools.py]: What do you mean?

Comment (lostella) [reply src/gluonts/itertools.py]: I mean, one could split whatever collection like this, not just a `Fuse` object

File: src/gluonts/itertools.py [hunk 3]
```diff
@@ -140,12 +145,150 @@ class Chain:
...
+        """Split into ``n`` parts of equal size."""
```
Comment (lostella) [inline src/gluonts/itertools.py]: Approximately equal

File: src/gluonts/itertools.py [hunk 4]
```diff
@@ -140,12 +145,150 @@ class Chain:
...
+
+        bucket_size, remainder = divmod(len(self), n)
+
+        # We need one fewer than `n`, since these become split positions.
+        relative_splits = np.full(bucket_size, n - 1)
+
+        # e.g. 100 by 3 -> 34, 33, 33
+        relative_splits[:r] += 1
```
Comment (lostella) [inline src/gluonts/itertools.py]: What is `r`?

**Review by lostella — CHANGES_REQUESTED** [review]
Tests! Otherwise looks good

File: src/gluonts/itertools.py (hunk 4 above)
Comment (jaheba) [reply src/gluonts/itertools.py]: oops

**Review by lostella — APPROVED** [review]

**Pull request merged**